test/RewardsStreamerMP.t.sol

```diff
@@ -1005,9 +1005,11 @@ contract StakeTest is RewardsStreamerMPTest {
         vm.warp(initialTime + 4 * YEAR);
         _stake(alice, stakeAmount, 0);
 
-        // staking with lock > 0 should revert as max lock time was reached
-        vm.expectRevert(IStakeManager.StakingManager__InvalidLockPeriod.selector);
-        _stake(alice, stakeAmount, MIN_LOCKUP_PERIOD);
+        _stake(alice, stakeAmount, 0);
+        // locking up to new limit should render same maxMP as staking initially the whole
+        _lock(alice, _lockTimeAvailable(stakeAmount * 4, streamer.getVault(vaults[alice]).maxMP));
```

Comment:
`_lockTimeAvailable` receives `balance` as first parameter, why are we sending `stakeAmount * 4`?

Reply:
@gravityblast Thanks for your comments.
1. I would not expect end users to have access to the error messages. We can rename the Error if you have a better naming, currently the error name is descriptive of its cause.

2. The goal of this PR is to allow users to lock tokens that was added after its already being locked. 
Without this PR, if you lock 100 tokens for 4 years, and later add more tokens, you wouldnt be able to lock your newly added tokens for the full period. 
The original requirements were that any token could generate up to 8 times its weight in multipliers (4 from bonus, 4 from accrual), which is the Absolute Max MP. Without this PR, tokens that were staked after a lock have been added wouldnt be able to reach this absolute max MP.